todo_react/todo-react/src/components/ToDoItem.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {useContext, useEffect, useRef, useState} from 'react';
+import {ITodo} from "../models";
+import {TodosContext} from "../context"
+interface TodoProps {
+    todo: ITodo
+}
+export function ToDoItem(props: TodoProps) {
+    const [isEditing, setIsEditing] = useState(false)
+    const [editableTitle, setEditableTitle] = useState<string>(props.todo.title)
+    const editTitleInputRef = useRef<HTMLInputElement>(null)
+    const todosContext = useContext(TodosContext)
```

Comment:
try archiving something like that
`const { editTodo, deleteTodo, doneTodo } = useTodosContext()`